client/landing/stepper/hooks/use-site-migraiton-key.ts

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+interface ApiResponse {
+	migration_key: string;
+}
+
+const getMigrationKey = async ( siteId: number ): Promise< ApiResponse > =>
+	wpcom.req.get( `/sites/${ siteId }/atomic-migration-status/migrate-guru-key?http_envelope=1`, {
+		apiNamespace: 'wpcom/v2',
+	} );
+
+export const useSiteMigrationKey = ( siteId?: number ) => {
+	return useQuery( {
+		queryKey: [ 'site-migration-key', siteId ],
+		queryFn: () => getMigrationKey( siteId! ),
+		retry: false,
+		enabled: !! siteId,
+		select: ( data ) => ( { migrationKey: data?.migration_key } ),
+		refetchOnWindowFocus: false,
+	} );
```

Comment:
Should we specify `staleTime` on the `useQuery()` call to allow the data to be kept in memory for a little while before we try to refetch? ([reference](https://tanstack.com/query/v4/docs/framework/react/reference/useQuery))

Reply:
I set 60 seconds as staleTime as we discussed on slack. 
https://github.com/Automattic/wp-calypso/pull/88952/commits/a8022e45bee59130156286d12da9f4d8ffe1a473